pkg/ipamd/ipamd.go

```diff
@@ -644,12 +651,109 @@ func (c *IPAMContext) updateLastNodeIPPoolAction() {
 	logPoolStats(total, used, c.maxIPsPerENI)
 }
 
-func (c *IPAMContext) tryAllocateENI() error {
+
+// For each ENIConfig we want to make sure there is at least
+// one matching interface attached
+func (c *IPAMContext) globalAllocateENI(enis []awsutils.ENIMetadata) error {
+
+	if c.eniConfig == nil {
+		return fmt.Errorf("eniConfig does not exist")
+	}
+
+	allENIConfigs := c.eniConfig.GetAllENIConfigs()
+
+	log.Debugf("Try to pre-allocate an ENI from each ENIConfig")
+
+	var existingENIs []v1alpha1.ENIConfigSpec
+	for _, eni := range enis {
+
+		var securityGroups []string
+
+		for _, sg := range eni.SecurityGroups {
+			securityGroups = append(securityGroups, *sg)
+		}
+
+		log.Debugf("Existing ENI %s, subnet %s, sgAll %s", eni.ENIID, eni.SubnetId, securityGroups)
+		existingENIs = append(existingENIs,v1alpha1.ENIConfigSpec{
+			SecurityGroups: securityGroups,
+			Subnet:         eni.SubnetId,
+		})
+	}
+	log.Debugf("existingENIs %s", existingENIs)
+
+	// TODO - This needs to filter down to just the ENIConfigs tagged to this specific host
+	// for now it configures for all ENIConfigs present
```

Comment:
Yes, we should test this on some large clusters.

Reply:
Are there existing tests to help facilitate this?

Reply:
And is additional filtering a merge blocker for this change?

Reply:
I don't think so, but I'll be sure to try running a custom network config setup on a somewhat large cluster to verify this first.